agent/connect/ca/provider_vault.go

```diff
@@ -551,13 +551,39 @@ func (v *VaultProvider) GenerateIntermediate() (string, error) {
 	}
 
 	// Set the intermediate backend to use the new certificate.
-	_, err = v.writeNamespaced(v.config.IntermediatePKINamespace, v.config.IntermediatePKIPath+"intermediate/set-signed", map[string]interface{}{
+	importResp, err := v.writeNamespaced(v.config.IntermediatePKINamespace, v.config.IntermediatePKIPath+"intermediate/set-signed", map[string]interface{}{
 		"certificate": intermediate.Data["certificate"],
 	})
 	if err != nil {
 		return "", err
 	}
 
+	if importResp != nil {
```

Comment:
Can Vault <1.11 also return a non-nil response? Wondering if we need to guard against POST-ing to `config/issuers` with empty string.

Reply:
Only [I believe](https://github.com/hashicorp/vault/blob/release/1.10.x/builtin/logical/pki/path_intermediate.go#L244) in [error cases](https://github.com/hashicorp/vault/blob/release/1.10.x/builtin/logical/pki/path_config_ca.go#L91). I'm not sure what `v.writeNamespaced` is, but in Vault's API client, we always return `(*vault.api.Secret, error)` -- the former cannot contain an error response. If you're directly using a lower-level API (one giving a raw response akin to `(vault.api.Response, error)`), you might end up with an error in the response object (e.g., if `logical.ErrorResponse` [is used on the server](https://github.com/hashicorp/vault/blob/release/1.10.x/builtin/logical/pki/path_intermediate.go#L204)), so you'd have to translate that to an `err` first.


_Assuming_ that has been done, and _assuming_ this is a Vault `*api.Secret` object, then the answer is no, and we strictly need to deal with non-nil responses here from `set-signed` on Vault 1.11+.